src/UIBook.elm

```diff
@@ -621,20 +633,26 @@ update msg model =
                         logAction_ ("Navigate to: " ++ url.path)
 
         OnUrlChange url ->
-            if url.path == "/" then
-                ( { model | chapterActive = Nothing }, Cmd.none )
+            case ( url.path, Array.get 0 model.chapters ) of
+                ( "/", Just chapter_ ) ->
+                    ( model
+                    , Nav.pushUrl model.navKey <| urlFromChapter model.config.urlPreffix chapter_
+                    )
 
-            else
-                let
-                    activeChapter =
-                        parseActiveChapterFromUrl model.config.urlPreffix model.chapters url
-                in
-                ( { model
-                    | chapterActive = activeChapter
-                    , isMenuOpen = False
-                  }
-                , maybeRedirect model.navKey activeChapter
-                )
+                ( "/", Nothing ) ->
+                    ( { model | chapterActive = Nothing }, Cmd.none )
+
+                _ ->
+                    let
+                        activeChapter =
+                            parseActiveChapterFromUrl model.config.urlPreffix model.chapters url
+                    in
+                    ( { model
+                        | chapterActive = activeChapter
+                        , isMenuOpen = False
+                      }
+                    , maybeRedirect model.navKey activeChapter
```

Comment:
Since `maybeRedirect` helper is only used here, we can make the code easier to understand by being more explicit:

```suggestion
                    , case activeChapter of
                        Just _ ->
                            Cmd.none

                        Nothing ->
                            Nav.replaceUrl model.navKey "/"
```

Then you can safely remove the `maybeRedirect` helper.